solr/core/src/java/org/apache/solr/security/PKIAuthenticationPlugin.java

```diff
@@ -485,34 +487,75 @@ private Optional<String> getUser() {
     }
   }
 
+  private static class CachedToken {
+    Instant generatedAt;
+    String token;
+
+    private CachedToken(Instant generatedAt, String token) {
+      this.generatedAt = generatedAt;
```

Comment:
not a big deal, but if we just store the expiration time instead of the generation time we don't have to subtract every time

Reply:
+1 to the above and maybe also add a method `isExpired()` to simplify the code